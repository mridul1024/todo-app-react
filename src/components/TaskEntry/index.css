.task-entry-container {
  margin-top: 20vh;
  background: rgba(255, 255, 255, 0.05);
  border-top: 5px solid #00526c;
  border-bottom: 5px solid #00526c;
  padding: 10px 0px;
  display: flex;
  align-items: center;
}

.task-entry-container input {
  margin-left: 30%;
  padding-right: 50px;
  width: 400px;
  font-size: 18px;
  background: transparent;
  border: none;
  color: rgba(255, 255, 255, 0.7);
}

.task-entry-container input:focus {
  outline: none;
}

.task-entry-container img {
  width: 30px;
  height: 30px;
  cursor: pointer;
}
