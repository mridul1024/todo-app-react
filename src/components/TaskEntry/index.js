import { React } from "react";
import "./index.css";

export default function TaskEntry({ InsertTask }) {
  return (
    <>
      <div className="task-entry-container">
        <input
          type="text"
          id="task-entry"
          data-testid="task-entry-test-id"
          placeholder="Write your new task here !"
        />
        <img
          onClick={InsertTask}
          data-testid="add-task-button-test-id"
          src="../add_task_button.png"
          alt="add_task_button"
        />
      </div>
    </>
  );
}
