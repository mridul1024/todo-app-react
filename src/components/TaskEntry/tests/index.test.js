import React from "react";
import { screen, render, fireEvent } from "@testing-library/react";
import TaskEntry from "../index";

describe("<TaskEntry /> component tests ->", () => {
  let clickSpy;
  beforeAll(() => {
    clickSpy = jest.fn();
  });
  test("should render the component and match snapshot", () => {
    //Arrange
    let taskEntryComponent;
    //Act
    taskEntryComponent = render(<TaskEntry InsertTask={clickSpy} />);
    //Assert
    expect(taskEntryComponent.baseElement).toMatchSnapshot("baseElement");
  });

  test("should contain a placeholder text", () => {
    //Arrange
    render(<TaskEntry InsertTask={clickSpy} />);
    let placeholder;
    //Act
    placeholder = screen.getByTestId("task-entry-test-id");
    //Assert
    expect(placeholder.hasAttribute("placeholder")).toBe(true);
  });

  test('should contain "Write your new task here !" as placeholder text value', () => {
    //Arrange
    render(<TaskEntry InsertTask={clickSpy} />);
    let taskEntryPlaceholder;
    //Act
    taskEntryPlaceholder = screen.getByTestId("task-entry-test-id");
    //Arrange
    expect(taskEntryPlaceholder.getAttribute("placeholder")).toBe(
      "Write your new task here !"
    );
  });

  test("should contain a task creation button", () => {
    //Arrange
    render(<TaskEntry InsertTask={clickSpy} />);
    //Act
    const button = screen.getAllByTestId("add-task-button-test-id");
    //Assert
    expect(button.length).toBe(1);
  });

  test("should call the prop InsertTask on button click", () => {
    //Arrange
    render(<TaskEntry InsertTask={clickSpy} />);
    //Act
    const button = screen.getByTestId("add-task-button-test-id");
    fireEvent.click(button);
    //Assert
    expect(clickSpy).toBeCalled();
  });
});
