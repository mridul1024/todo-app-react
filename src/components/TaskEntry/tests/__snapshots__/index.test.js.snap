// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskEntry /> component tests -> should render the component and match snapshot: baseElement 1`] = `
<body>
  <div>
    <div
      class="task-entry-container"
    >
      <input
        data-testid="task-entry-test-id"
        id="task-entry"
        placeholder="Write your new task here !"
        type="text"
      />
      <img
        alt="add_task_button"
        data-testid="add-task-button-test-id"
        src="../add_task_button.png"
      />
    </div>
  </div>
</body>
`;
